sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -359,29 +360,32 @@ def _from_connection_string(conn_str: str, **kwargs: Any) -> Dict[str, Any]:
             kwargs["credential"] = EventHubSharedKeyCredential(policy, key)
         return kwargs
 
-    def _create_auth(self) -> Union["uamqp_JWTTokenAuth", JWTTokenAuth]:
+    def _create_auth(self, *, auth_uri: Optional[str] = None) -> Union["uamqp_JWTTokenAuth", JWTTokenAuth]:
         """
         Create an ~uamqp.authentication.SASTokenAuth instance
          to authenticate the session.
 
         :return: The auth for the session.
         :rtype: JWTTokenAuth or uamqp_JWTTokenAuth
         """
+        # if auth_uri is not provided, use the default hub one
+        entity_auth_uri = auth_uri if auth_uri else self._eventhub_auth_uri
```

Comment:
will there be a case there where we dont provide auth_uri ?

Reply:
We always pass an auth URI into the token, but we don't pass auth_uri when creating auth for the MgmtClient. It'll be this default eventhub_auth_uri value. I could just make this a required property and pass in the eventhub_auth_uri explicitly in the MgmtClient operations. Not sure if you prefer this or the other.

Reply:
no preference, the kwarg is fine. I was just wondering if the producer also needs the new url ?

Reply:
it does not, since sending happens to the hub and not a consumer group. Consumer group should just be for receiving.